cmd/pebble/cmd_enter.go

```diff
@@ -0,0 +1,188 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/canonical/pebble/internal/logger"
+	"github.com/jessevdk/go-flags"
+)
+
+const shortEnterHelp = "Run pebble command in new pebble environment"
+const longEnterHelp = `
+The enter command behaves like the run command with the possibility of running
+a single pebble command after starting the service manager. It's intended to be
+used as an entrypoint in container images.
+
+When no subcommand is given, the behavior is identical to the run command.
+
+When a subcommand is given, the service manager is started in the background
+without starting "autostart" services (i.e. run --hold, unless --run flag is
+given), and the subcommand is run as a pebble client command that can
+communicate with the service manager instance in the background. The service
+manager is shut down after the commands finish unless noted otherwise below.
+
+Only a subset of pebble commands is supported as a subcommand:
+
+  exec
+  help
+  ls
+  plan
+  services
+  start
+  version
+
+Furthermore, some commands have restrictions and idiosyncrasies:
+
+  exec, ls, plan, services, version
+    Service manager logging is disabled unless -v/--verbose flag is provided.
+
+  help, ls, plan, services, version
+    Service autostart (the --run flag) is prohibited.
+
+  start, stop
+    Service manager continues running after the command finishes.
+
+  stop
+    Service autostart (the --run flag) is required.
```

Comment:
We forgot to address this wart in our call yesterday. We need to figure why start/stop were proposed to be different, as it's very awkward to require a flag at all times for stop and not for start.

Reply:
Recalling the spec (RK007) discussions, the logic was that `pebble enter stop [svc ...]` would have an equivalent behaviour to `pebble run --hold; pebble stop [svc ...]`, which was deemed to be undesired and thus marked as an "error" combination. 

> Service autostart (the --run flag) is required.

Putting this in order words, `--run` is required as a consequence of `pebble enter` acting as `pebble run --hold`. Without it,  `enter stop` doesn't do anything cause there are no "started" services.